appmap/_implementation/testing_framework.py

```diff
@@ -0,0 +1,123 @@
+import inflection
+import re
+
+from contextlib import contextmanager
+from pathlib import Path
+
+from appmap._implementation import configuration, env, generation, recording
+from appmap._implementation.utils import fqname
+
+
+try:
+    # Make sure we have hooks installed if we're testing a django app.
+    # pylint: disable=unused-import
+    import appmap.django  # noqa: F401
+except ImportError:
+    pass  # probably no django
+
+
+class FuncItem:
+    def __init__(self, cls, name, method_id=None):
+        self.cls = cls
+        self.name = name
+        self.method_id = method_id or name
+
+    @property
+    def class_name(self):
+        return self.cls.__name__
+
+    @property
+    def defined_class(self):
+        return fqname(self.cls)
+
+    def is_in_class(self):
+        return self.cls is not None
+    has_feature_group = is_in_class
+
+    @property
+    def feature_group(self):
+        test_name = self.class_name
+        if test_name.startswith('Test'):
+            test_name = test_name[4:]
+        return inflection.humanize(inflection.titleize(test_name))
+
+    @property
+    def feature(self):
+        return inflection.humanize(self.test_name)
+
+    @property
+    def scenario_name(self):
+        name = '%s%s' % (self.feature[0].lower(), self.feature[1:])
+        if self.has_feature_group():
+            name = ' '.join([self.feature_group, name])
+        return name
+
+    @property
+    def test_name(self):
+        ret = self.name
+        return re.sub('^test.?_', '', ret)
+
+    @property
+    def filename(self):
+        fname = self.name
+        if self.cls:
+            fname = '%s_%s' % (self.defined_class, fname)
+        fname = re.sub('[^a-zA-Z0-9-_]', '_', fname)
+        if fname.endswith('_'):
+            fname = fname[:-1]
+        return fname
+
+    @property
+    def metadata(self):
+        ret = {}
+        if self.is_in_class():
+            ret['feature_group'] = self.feature_group
+            ret['recording'] = {
+                'defined_class': self.defined_class,
+                'method_id': self.method_id
+            }
+        ret.update({
+            'name': self.scenario_name,
+            'feature': self.feature
+        })
+        return ret
+
+
+class session:
+    def __init__(self, name, version=None):
+        if not env.enabled():
+            return
+
+        self.appmap_path = Path(configuration.Config().output_dir) / name
+        self.appmap_path.mkdir(parents=True, exist_ok=True)
+
+        # the tests might chdir so make sure to keep the path absolute
+        self.appmap_path = self.appmap_path.resolve()
```

Comment:
As of #44, Config.output_dir is guaranteed to be an absolute path, so you don't need to do this.

(If somehow you're seeing different behavior, that's a bug, and we should fix it.)

Reply:
Ah, ok, thanks.